src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -0,0 +1,83 @@
+import datetime
+from typing import Union
+
+from sentry.exceptions import InvalidConfiguration
+from sentry.utils import redis
+
+from . import base
+
+
+class RedisRealtimeMetricsStore(base.RealtimeMetricsStore):
+    """An implementation of RealtimeMetricsStore based on a Redis backend."""
+
+    def __init__(
+        self,
+        cluster: str,
+        counter_bucket_size: int,
+        counter_ttl: datetime.timedelta,
+        histogram_bucket_size: int,
+        histogram_ttl: datetime.timedelta,
+    ) -> None:
+        """Creates a RedisRealtimeMetricsStore.
+
+        "cluster" is the name of the Redis cluster to use. "counter_bucket_size" is the size
+        in second of the buckets that timestamps will be sorted into when a project's event counter is incremented.
+        "counter_ttl" is the duration that event counter entries will be kept around for *after
+        the last increment call*.
+
+        "histogram_bucket_size" and "histogram_ttl" function like their "counter*" siblings,
+        but for the histograms.
+        """
+
+        self.cluster = redis.redis_clusters.get(cluster)
+        self._counter_bucket_size = counter_bucket_size
+        self._counter_ttl = int(counter_ttl / datetime.timedelta(milliseconds=1))
+        self._histogram_bucket_size = histogram_bucket_size
+        self._histogram_ttl = int(histogram_ttl / datetime.timedelta(milliseconds=1))
+        self._prefix = "symbolicate_event_low_priority"
+
+    def validate(self) -> None:
+        if self._counter_bucket_size <= 0:
+            raise InvalidConfiguration("bucket size must be at least 1")
```

Comment:
i left a comment asking about the existence of this in the counter PR, but if this is needed should the histogram bucket size also be checked here?